lib/drives/util/recurrence.dart

```diff
@@ -0,0 +1,259 @@
+    switch (intervalType!) {
+      case RecurrenceIntervalType.days:
+        return startedAt.add(Duration(days: intervalSize!));
+      case RecurrenceIntervalType.weeks:
+        return startedAt.add(Duration(days: intervalSize! * 7));
+      case RecurrenceIntervalType.months:
+        final int nextMonth = startedAt.month + intervalSize!;
+        return DateTime(startedAt.year + (nextMonth + 1) ~/ 12, (nextMonth + 1) % 12 - 1, startedAt.day);
+      case RecurrenceIntervalType.years:
+        return DateTime(startedAt.year + intervalSize!, startedAt.month, startedAt.day);
+    }
+  }
+
+  @override
+  String getName(BuildContext context) {
+    switch (intervalType!) {
+      case RecurrenceIntervalType.days:
+        return S.of(context).recurrenceEndForDays(intervalSize!);
+      case RecurrenceIntervalType.weeks:
+        return S.of(context).recurrenceEndForWeeks(intervalSize!);
+      case RecurrenceIntervalType.months:
+        return S.of(context).recurrenceEndForMonths(intervalSize!);
+      case RecurrenceIntervalType.years:
+        return S.of(context).recurrenceEndForYears(intervalSize!);
+    }
+  }
+
+  @override
+  String? validate(BuildContext context) {
+    if (intervalSize == null) return S.of(context).recurrenceEndForValidationIntervalSizeNull;
+    if (intervalType == null) return S.of(context).recurrenceEndForValidationIntervalTypeNull;
+    if (intervalType == RecurrenceIntervalType.years && intervalSize! >= 10 || intervalSize! >= 100) {
+      return S.of(context).recurrenceEndForValidationIntervalTooLarge;
+    }
+    return null;
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      other is RecurrenceEndChoiceInterval &&
+      isCustom == other.isCustom &&
+      intervalSize == other.intervalSize &&
+      intervalType == other.intervalType;
+
+  @override
+  RecurrenceEndChoice copyWith({bool? isCustom}) =>
+      RecurrenceEndChoiceInterval(intervalSize, intervalType, isCustom: isCustom ?? this.isCustom);
+
+  @override
+  int get hashCode => isCustom.hashCode ^ intervalSize.hashCode ^ intervalType.hashCode;
+}
+
+class RecurrenceEndChoiceOccurrence extends RecurrenceEndChoice {
+  int? occurrences;
+
+  RecurrenceEndChoiceOccurrence(this.occurrences, {super.isCustom})
+      : assert(occurrences != null || isCustom),
+        super(type: RecurrenceEndType.occurrence);
+
+  @override
+  String getName(BuildContext context) => S.of(context).recurrenceEndAfterOccurrences(occurrences!);
+
+  @override
+  String? validate(BuildContext context) => occurrences == null
+      ? S.of(context).recurrenceEndOccurrencesValidationNull
+      : occurrences! >= 100
+          ? S.of(context).recurrenceEndValidationOccurrencesTooMany
+          : null;
+
+  @override
+  bool operator ==(Object other) =>
+      other is RecurrenceEndChoiceOccurrence && isCustom == other.isCustom && occurrences == other.occurrences;
+
+  @override
+  RecurrenceEndChoice copyWith({bool? isCustom}) =>
```

Comment:
This is not covered, maybe it's not even needed anymore?

Reply:
Yeah I saw that... Wasn't that the one where some arguments were never called? I thought I'd leave that for consistency. But there are some places in the code where I think some parameters/methods are not needed, lmk if you think I should just delete those